### PR TITLE
Update docs to mention AMR support for mattermost plugin

### PR DESCRIPTION
**Pull request by EdwardDowling**
Part of [42856](https://github.com/gravitational/teleport/issues/42856)

### CODE REVIEW

File: docs/pages/access-controls/access-request-plugins/notification-routing-rules.mdx [hunk 1]
```diff
@@ -11,7 +11,7 @@ For example, you may wish to send notifications to different Slack channels
 based on what roles or resources are being accessed, or based on what user
 is making the request.
 
-Currently only the Slack access plugin is supported.
+Currently only the Slack and Mattermost access plugins supported.
```
Comment (r0mant) [inline docs/pages/access-controls/access-request-plugins/notification-routing-rules.mdx:14]: As we're working on enabling them can we rephrase this to avoid changing this every time you add a new plugin? To something like a note-style admonition that says `Currently, only a subset of hosted plugins support notification routing rules. We are working on extending support to the rest of hosted plugins. Keep an eye on Teleport changelog to learn about new plugins being added.` with a link to the changelog page.

File: docs/pages/access-controls/access-request-plugins/notification-routing-rules.mdx [hunk 2]
```diff
@@ -11,7 +11,11 @@ For example, you may wish to send notifications to different Slack channels
 based on what roles or resources are being accessed, or based on what user
 is making the request.
 
-Currently only the Slack access plugin is supported.
+Currently, only a subset of hosted plugins support notification routing rules.
+We are working on extending support to the rest of hosted plugins. Keep an eye
+on Teleport changelog to learn about new plugins being added.
+
+[Changelog](../../changelog.mdx)
```
Comment (r0mant) [inline docs/pages/access-controls/access-request-plugins/notification-routing-rules.mdx:18]:
```suggestion
on Teleport [changelog](../../changelog.mdx) to learn about new plugins being added.
```

**Review by r0mant — APPROVED** [review]

File: docs/pages/access-controls/access-request-plugins/notification-routing-rules.mdx [hunk 3]
```diff
@@ -11,7 +11,11 @@ For example, you may wish to send notifications to different Slack channels
 based on what roles or resources are being accessed, or based on what user
 is making the request.
 
-Currently only the Slack access plugin is supported.
+Currently, only a subset of hosted plugins support notification routing rules.
+We are working on extending support to the rest of hosted plugins. Keep an eye
+on Teleport changelog to learn about new plugins being added.
```
Comment (ptgott) [inline docs/pages/access-controls/access-request-plugins/notification-routing-rules.mdx:16]:
```suggestion
We are working on extending support to the rest of hosted plugins. Keep an eye
on the Teleport changelog to learn about new plugins being added.
```

Article use. See: https://www.grammarly.com/blog/articles/

**Review by ptgott — APPROVED** [review]
